### ISSUE TITLE
Can't get prepan to see my github mail

### ISSUE THREAD

**Issue by kenneth-olwing** [metacpan/prepan#55]
Hi,

No matter what I seem to try, when looking at "Edit profile" I keep getting:

  _We couldn't retrieve your email address. If you want to be notified via email, set at least 1 email
  address visibile at your setting page on GitHub. Then sign in PrePAN again._

AFAICT, I have a public github mail adress set so...not sure what I'm missing...?

Thank you,

ken1

**Comment by shibayu36** [comment]
I visited your profile page (https://github.com/kenneth-olwing), but I could not find your public email address.

Do you setup your github email address correctly? In my email setting page ( https://github.com/settings/emails ), my setting is following.

![image](https://cloud.githubusercontent.com/assets/1132489/17174350/df9cc2e4-543b-11e6-8827-d926297e272d.png)

**Comment by kenneth-olwing** [comment]
Hi,
Thanks for checking - but yes, I have three adresses there and one of them (github@olwing.se) looks like that i.e. is marked primary/public. At first I did have just one marked private, but the other two was normal to add, I could verify them etc, and once that was done I could switch between which was supposed to be primary, and unchecking the 'keep private' box. I can't find any other reasonable setting that should influence this...just weird???

**Comment by shibayu36** [comment]
fmm... I cannot find the reason why prepan cannot get your email address...

I haven't deployed prepan application recently, so there is a possibility that prepan cannot get email address from github because scope settings of github OAuth has been changed.

**Comment by kenneth-olwing** [comment]
Thanks for trying :-)

I don't know much about OAuth, but looking at that settings page there seems little it says other than 'I allow Prepan to access my account'.

Anyway, that still doesn't explain why my mail isn't visible on my page...I just looked on another machine where the browser have never known 'me' and you're correct - the mail adress isn't visible.

I'll look into it with github support I guess and get back here when I know more!

**Comment by kenneth-olwing** [comment]
...and just like that I found the missing piece...

It's not enough to mark a mail adress as public on the Personal settings/Emails page - there's also a 'Public email' setting on the Personal settings/Profile page. In fact, I'm not sure that it matters how the mail adresses are marked on the first page, I haven't experimented...

Doh. I thought I was looking in the pertinent places, but I guess it was too much to assume that the necessary setting was only on the **_Emails_** setting page...:-). Sorry for the confusion!

As a good thing maybe you'd consider to update prepan so the error message I started out with links to the profile page too (or at least mentions that it's important to adjust that too - or instead).

Thanks for doing your bit in any case; ultimately I dropped the ball. Have a nice one!
